stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -304,9 +304,11 @@ public struct Unsafe${Mutable}BufferPointer<Element>
 
       // FIXME: swift-3-indexing-model: tests.
       if !newValue.isEmpty {
-        (_position! + bounds.lowerBound).assign(
-          from: newValue.base._position! + newValue.startIndex,
-          count: newValue.count)
+        let source = newValue.base._position! + newValue.startIndex
+        let destination = _position! + bounds.lowerBound
+        if source != destination {
```

Comment:
If it's appropriate to do this check, is there a reason not to do it directly as an `else if` in `assign`?
https://github.com/apple/swift/blob/e4236405279f42345e12eebbc023a3409a034b7a/stdlib/public/core/UnsafePointer.swift.gyb#L562-L564

Reply:
or even all the way down in `array_copy_operation`?
https://github.com/apple/swift/blob/e4236405279f42345e12eebbc023a3409a034b7a/stdlib/public/runtime/Array.cpp#L86-L87